### PR TITLE
Update captive-portal spec

### PR DESCRIPTION
**Pull request by joelanders**
I looked around for what Microsoft, Google, and Apple do for captive portal detection these days and wrote a test for it in measurement-kit here:
https://github.com/measurement-kit/measurement-kit/pull/1238

I'm proposing a different report format here.

### CODE REVIEW

**Comment by bassosimone** [comment]
I am mostly okay with these changes. What I am wondering is whether it would perhaps be possible to keep similar keys to the previous version, so to have a common parser that is able to parse either version 0.3 or version 0.4 (equal to 0.3 but with more keys).

Another aspect that I think should be emphasized is that you say you have checked what is currently being done *nowadays* by the vendors. Since this seems to be different from what we were doing previously, I was thinking that perhaps you should document the sources from which you did ready what they do, so they are here for the records.

I'd also like to see whether @hellais, also pinged [here](https://github.com/measurement-kit/measurement-kit/pull/1272#issuecomment-313447783) has further comments.

**Comment by hellais** [comment]
Why was this PR closed?

**Comment by hellais** [comment]
I think the main point of friction is whether or not we can preserve some minimal amount of backward compatibility for the test or if it's necessary to just scratch everything and re-do it from scratch.

**Comment by joelanders** [comment]
Ah, I was going to re-open another PR here that just added citations for the methodology. I think we can keep the report format (working on that now).

**Comment by joelanders** [comment]
Ok, never mind, some key names have to be added, so let's keep this PR open and I'll modify it.

**Comment by willscott** [comment]
@bassosimone is this reflecting the current state of MK (the v0.4 of the spec in this PR claims to explain MK behavior)? if so, we should merge this

File: test-specs/ts-010-captive-portal.md [hunk 1]
```diff
@@ -16,13 +16,44 @@ The ability to detect a captive portal.
...
+## Version 0.3.0 and earlier:
+
 The test is split in four smaller tests:
 
     1. Http test that compares a requested document against the control one (vendor_tests).
     2. DNS test that checks a NXDOMAIN response to random hostnames requests (part of the vendor_dns_tests).
     3. DNS test that compares an A request to a static domain against its IP (part of the vendor_dns_tests).
     4. DNS test that check for 0x20 encoding (check0x20 tests).
 
+## Version 0.4.0 (currently only in measurement-kit):
```
Comment (hellais) [inline test-specs/ts-010-captive-portal.md]: I believe this is currently not implemented. Probably we should change this bit of text and say that this is work in progress.

Comment (hellais) [reply test-specs/ts-010-captive-portal.md]: Actually this is not true. However we are not running it from the mobile app and it's still missing some features imho. I documented those here: https://github.com/measurement-kit/measurement-kit/issues/1577

**Comment by hellais** [comment]
I updated the section header and am merging this.

**Pull request merged**